src/traces/scattergl/index.js

```diff
@@ -326,7 +326,15 @@ function plot(gd, subplot, cdata) {
     var width = fullLayout.width;
     var height = fullLayout.height;
 
-    prepareRegl(gd, ['ANGLE_instanced_arrays', 'OES_element_index_uint']);
+    var success = prepareRegl(gd, ['ANGLE_instanced_arrays', 'OES_element_index_uint']);
+    if(!success) {
+        scene.error2d = false;
+        scene.line2d = false;
+        scene.scatter2d = false;
+        scene.fill2d = false;
```

Comment:
what happens if we add some other component to support a new feature, would we also have to add it here? Will the tests below fail and force that, or would we need to add a new test for the new feature? Just wondering if there's some way we can preempt these concerns like
```js
if(!success) {
  for(var key in scene) {
    if(something about key or scene.key) scene[key] = false;
  }
}
```

Reply:
all DRY'ed up in eb594d1bf96d0640b5fe7d5b75aac67f3e2418a0